tests/react/async.test.tsx

```diff
@@ -968,21 +968,36 @@ it('async atom double chain without setTimeout (#751)', async () => {
 
 it('async atom double chain with setTimeout', async () => {
   const enabledAtom = atom(false)
-  const resolve: (() => void)[] = []
+  const resolveMap = new WeakMap<Atom<unknown>, (() => void)[]>()
+  async function addResolve(a: Atom<unknown>) {
+    const promiseArray = resolveMap.get(a)
+    if (!promiseArray) {
+      await new Promise<void>((r) => resolveMap.set(a, [r]))
+    } else {
+      await new Promise<void>((r) => promiseArray.push(r))
+    }
+  }
```

Comment:
Just for preference:
```suggestion
  function addResolve(a: Atom<unknown>) {
    const promiseArray = resolveMap.get(a)
    if (!promiseArray) {
      return new Promise<void>((r) => resolveMap.set(a, [r]))
    }
    return new Promise<void>((r) => promiseArray.push(r))
  }
```